File: theworldcore/Controllers/Api/TripsController.cs
using System.Threading.Tasks;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using theworldcore.Model;

namespace theworldcore.Controllers.Api
{
    [Authorize]
    [Route("api/trips")]
    public class TripsController: Controller
    {
        private readonly IWorldRepository repository;

        public TripsController(IWorldRepository repository)
        {
            this.repository = repository;
        }

        [HttpGet]
        public async Task<IActionResult> Get()
        {
           await Task.Delay(1000);
           return Ok(repository.GetTripsByUsername(User.Identity.Name));
        }


        [HttpPost]
        public async Task<IActionResult> Post([FromBody]Trip newTrip)
        {
            if (ModelState.IsValid)
            {
                newTrip.UserName = User.Identity.Name;
                repository.AddTrip(newTrip);
                await repository.SaveChangesAsync();

                await Task.Delay(1000);
                return Created($"api/trips/{newTrip.Id}", newTrip);
            }

            return BadRequest("Failed to save trip");
        }
    }
}

File: theworldcore/wwwroot/js/tripsController.js
(function () {

    angular.module("app-trips").controller("tripsController", tripsController);

    function tripsController($http) {
        var vm = this;
        vm.errorMessage = "";
        vm.isBusy = true;
        vm.newTrip = {};

        $http.get('/api/trips')
            .then(function (response) {
                vm.trips = angular.copy(response.data);
            })
            .catch(function (e) {
                vm.errorMessage = "Failed to load data:" + e.statusText;
            }).finally(function () {
                vm.isBusy = false;
            });




        vm.addTrip = function () {
            vm.errorMessage = "";
            vm.isBusy = true;

            $http.post('/api/trips', vm.newTrip)
                .then(function (response) {
                    vm.trips.push(vm.newTrip);
                })
                .catch(function (e) {
                    vm.errorMessage = "Failed to add new trip data:" + e.statusText;
                }).finally(function () {
                    vm.isBusy = false;
                    vm.newTrip = {};

                });

        }
    }

})();